compiler/sigmatch.nim

```diff
@@ -1344,6 +1344,8 @@ proc typeRel(c: var TCandidate, f, aOrig: PType,
     elif a.kind == tyNil: result = f.allowsNil
     else: discard
   of tyProc:
+    if a.kind == tyProc and a.owner != nil and a.owner.kind in {skTemplate, skMacro}:
+      return isNone
```

Comment:
The manual includes **no** special rules for overload resolution for templates and macros! I know that the current codebase disagrees with that but please don't make it worse.